// ionic-angular/components/searchbar/searchbar.ngfactory.ts
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
 /* tslint:disable */

import * as import0 from '@angular/core/src/render/api';
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/searchbar/searchbar';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import5 from '@angular/core/src/di/injector';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/config/config';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/forms/src/directives/ng_control';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from '@angular/core/src/linker/component_factory';
import * as import13 from '@angular/core/src/linker/query_list';
import * as import14 from 'ionic-angular/components/button/button';
import * as import15 from 'ionic-angular/components/icon/icon';
import * as import16 from '@angular/forms/src/directives/default_value_accessor';
import * as import17 from '@angular/forms/src/directives/ng_model';
import * as import18 from '@angular/forms/src/directives/ng_control_status';
import * as import19 from '../button/button.ngfactory';
import * as import20 from '@angular/forms/src/directives/control_value_accessor';
var renderType_Searchbar_Host:import0.RenderComponentType = (null as any);
class _View_Searchbar_Host0 extends import1.AppView<any> {
  _el_0:any;
  /*private*/ _appEl_0:import2.AppElement;
  _Searchbar_0_4:import3.Searchbar;
  /*private*/ _expr_0:any;
  /*private*/ _expr_1:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_Searchbar_Host0,renderType_Searchbar_Host,import6.ViewType.HOST,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    this._el_0 = this.selectOrCreateHostElement('ion-searchbar',rootSelector,(null as any));
    this._appEl_0 = new import2.AppElement(0,(null as any),this,this._el_0);
    var compView_0:any = viewFactory_Searchbar0(this.viewUtils,this.injector(0),this._appEl_0);
    this._Searchbar_0_4 = new import3.Searchbar(this.parentInjector.get(import8.Config),new import9.ElementRef(this._el_0),this.renderer,this.parentInjector.get(import10.NgControl,(null as any)));
    this._appEl_0.initComponent(this._Searchbar_0_4,[],compView_0);
    compView_0.create(this._Searchbar_0_4,this.projectableNodes,(null as any));
    this._expr_0 = import7.UNINITIALIZED;
    this._expr_1 = import7.UNINITIALIZED;
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this.init([].concat([this._el_0]),[this._el_0],[],[]);
    return this._appEl_0;
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import3.Searchbar) && (0 === requestNodeIndex))) { return this._Searchbar_0_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    if (((this.numberOfChecks === 0) && !throwOnChange)) { this._Searchbar_0_4.ngOnInit(); }
    this.detectContentChildrenChanges(throwOnChange);
    const currVal_0:any = this._Searchbar_0_4._value;
    if (import4.checkBinding(throwOnChange,this._expr_0,currVal_0)) {
      this.renderer.setElementClass(this._el_0,'searchbar-has-value',currVal_0);
      this._expr_0 = currVal_0;
    }
    const currVal_1:any = this._Searchbar_0_4._isActive;
    if (import4.checkBinding(throwOnChange,this._expr_1,currVal_1)) {
      this.renderer.setElementClass(this._el_0,'searchbar-active',currVal_1);
      this._expr_1 = currVal_1;
    }
    const currVal_2:any = this._Searchbar_0_4.showCancelButton;
    if (import4.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this.renderer.setElementClass(this._el_0,'searchbar-show-cancel',currVal_2);
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = this._Searchbar_0_4.shouldAlignLeft();
    if (import4.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this.renderer.setElementClass(this._el_0,'searchbar-left-aligned',currVal_3);
      this._expr_3 = currVal_3;
    }
    const currVal_4:any = this._Searchbar_0_4._sbHasFocus;
    if (import4.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this.renderer.setElementClass(this._el_0,'searchbar-has-focus',currVal_4);
      this._expr_4 = currVal_4;
    }
    this.detectViewChildrenChanges(throwOnChange);
    if (!throwOnChange) { this._Searchbar_0_4.ngAfterViewChecked(); }
  }
}
function viewFactory_Searchbar_Host0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<any> {
  if ((renderType_Searchbar_Host === (null as any))) { (renderType_Searchbar_Host = viewUtils.createRenderComponentType('',0,import11.ViewEncapsulation.None,[],{})); }
  return new _View_Searchbar_Host0(viewUtils,parentInjector,declarationEl);
}
export const SearchbarNgFactory:import12.ComponentFactory<import3.Searchbar> = new import12.ComponentFactory<import3.Searchbar>('ion-searchbar',viewFactory_Searchbar_Host0,import3.Searchbar);
const styles_Searchbar:any[] = [];
var renderType_Searchbar:import0.RenderComponentType = (null as any);
class _View_Searchbar0 extends import1.AppView<import3.Searchbar> {
  _viewQuery_searchbarInput_0:import13.QueryList<any>;
  _viewQuery_searchbarIcon_1:import13.QueryList<any>;
  _viewQuery_cancelButton_2:import13.QueryList<any>;
  _el_0:any;
  _el_1:any;
  /*private*/ _appEl_1:import2.AppElement;
  _Button_1_4:import14.Button;
  _el_2:any;
  _Icon_2_3:import15.Icon;
  _el_3:any;
  _el_4:any;
  _DefaultValueAccessor_4_3:import16.DefaultValueAccessor;
  _NG_VALUE_ACCESSOR_4_4:any[];
  _NgModel_4_5:import17.NgModel;
  _NgControl_4_6:any;
  _NgControlStatus_4_7:import18.NgControlStatus;
  _el_5:any;
  /*private*/ _appEl_5:import2.AppElement;
  _Button_5_4:import14.Button;
  _el_6:any;
  /*private*/ _appEl_6:import2.AppElement;
  _Button_6_4:import14.Button;
  _text_7:any;
  /*private*/ _expr_2:any;
  /*private*/ _expr_3:any;
  /*private*/ _expr_4:any;
  /*private*/ _expr_5:any;
  /*private*/ _expr_12:any;
  /*private*/ _expr_13:any;
  /*private*/ _expr_14:any;
  /*private*/ _expr_15:any;
  /*private*/ _expr_16:any;
  /*private*/ _expr_17:any;
  /*private*/ _expr_18:any;
  /*private*/ _expr_19:any;
  /*private*/ _expr_22:any;
  /*private*/ _expr_25:any;
  /*private*/ _expr_26:any;
  /*private*/ _expr_27:any;
  constructor(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement) {
    super(_View_Searchbar0,renderType_Searchbar,import6.ViewType.COMPONENT,viewUtils,parentInjector,declarationEl,import7.ChangeDetectorStatus.CheckAlways);
  }
  createInternal(rootSelector:string):import2.AppElement {
    const parentRenderNode:any = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
    this._viewQuery_searchbarInput_0 = new import13.QueryList<any>();
    this._viewQuery_searchbarIcon_1 = new import13.QueryList<any>();
    this._viewQuery_cancelButton_2 = new import13.QueryList<any>();
    this._el_0 = this.renderer.createElement(parentRenderNode,'div',(null as any));
    this.renderer.setElementAttribute(this._el_0,'class','searchbar-input-container');
    this._el_1 = this.renderer.createElement(this._el_0,'button',(null as any));
    this.renderer.setElementAttribute(this._el_1,'class','searchbar-md-cancel');
    this.renderer.setElementAttribute(this._el_1,'clear','');
    this.renderer.setElementAttribute(this._el_1,'color','dark');
    this.renderer.setElementAttribute(this._el_1,'ion-button','');
    this._appEl_1 = new import2.AppElement(1,0,this,this._el_1);
    var compView_1:any = import19.viewFactory_Button0(this.viewUtils,this.injector(1),this._appEl_1);
    this._Button_1_4 = new import14.Button((null as any),'',this.parentInjector.get(import8.Config),new import9.ElementRef(this._el_1),this.renderer);
    this._appEl_1.initComponent(this._Button_1_4,[],compView_1);
    this._el_2 = this.renderer.createElement((null as any),'ion-icon',(null as any));
    this.renderer.setElementAttribute(this._el_2,'name','arrow-back');
    this.renderer.setElementAttribute(this._el_2,'role','img');
    this._Icon_2_3 = new import15.Icon(this.parentInjector.get(import8.Config),new import9.ElementRef(this._el_2),this.renderer);
    compView_1.create(this._Button_1_4,[[].concat([this._el_2])],(null as any));
    this._el_3 = this.renderer.createElement(this._el_0,'div',(null as any));
    this.renderer.setElementAttribute(this._el_3,'class','searchbar-search-icon');
    this._el_4 = this.renderer.createElement(this._el_0,'input',(null as any));
    this.renderer.setElementAttribute(this._el_4,'class','searchbar-input');
    this._DefaultValueAccessor_4_3 = new import16.DefaultValueAccessor(this.renderer,new import9.ElementRef(this._el_4));
    this._NG_VALUE_ACCESSOR_4_4 = [this._DefaultValueAccessor_4_3];
    this._NgModel_4_5 = new import17.NgModel((null as any),(null as any),(null as any),this._NG_VALUE_ACCESSOR_4_4);
    this._NgControl_4_6 = this._NgModel_4_5;
    this._NgControlStatus_4_7 = new import18.NgControlStatus(this._NgControl_4_6);
    this._el_5 = this.renderer.createElement(this._el_0,'button',(null as any));
    this.renderer.setElementAttribute(this._el_5,'class','searchbar-clear-icon');
    this.renderer.setElementAttribute(this._el_5,'clear','');
    this.renderer.setElementAttribute(this._el_5,'ion-button','');
    this._appEl_5 = new import2.AppElement(5,0,this,this._el_5);
    var compView_5:any = import19.viewFactory_Button0(this.viewUtils,this.injector(5),this._appEl_5);
    this._Button_5_4 = new import14.Button((null as any),'',this.parentInjector.get(import8.Config),new import9.ElementRef(this._el_5),this.renderer);
    this._appEl_5.initComponent(this._Button_5_4,[],compView_5);
    compView_5.create(this._Button_5_4,[[]],(null as any));
    this._el_6 = this.renderer.createElement(parentRenderNode,'button',(null as any));
    this.renderer.setElementAttribute(this._el_6,'class','searchbar-ios-cancel');
    this.renderer.setElementAttribute(this._el_6,'clear','');
    this.renderer.setElementAttribute(this._el_6,'ion-button','');
    this._appEl_6 = new import2.AppElement(6,(null as any),this,this._el_6);
    var compView_6:any = import19.viewFactory_Button0(this.viewUtils,this.injector(6),this._appEl_6);
    this._Button_6_4 = new import14.Button((null as any),'',this.parentInjector.get(import8.Config),new import9.ElementRef(this._el_6),this.renderer);
    this._appEl_6.initComponent(this._Button_6_4,[],compView_6);
    this._text_7 = this.renderer.createText((null as any),'',(null as any));
    compView_6.create(this._Button_6_4,[[].concat([this._text_7])],(null as any));
    var disposable_0:Function = this.renderer.listen(this._el_1,'click',this.eventHandler(this._handle_click_1_0.bind(this)));
    var disposable_1:Function = this.renderer.listen(this._el_1,'mousedown',this.eventHandler(this._handle_mousedown_1_1.bind(this)));
    this._expr_2 = import7.UNINITIALIZED;
    this._expr_3 = import7.UNINITIALIZED;
    this._expr_4 = import7.UNINITIALIZED;
    this._expr_5 = import7.UNINITIALIZED;
    this._expr_12 = import7.UNINITIALIZED;
    var disposable_2:Function = this.renderer.listen(this._el_4,'ngModelChange',this.eventHandler(this._handle_ngModelChange_4_0.bind(this)));
    var disposable_3:Function = this.renderer.listen(this._el_4,'input',this.eventHandler(this._handle_input_4_1.bind(this)));
    var disposable_4:Function = this.renderer.listen(this._el_4,'blur',this.eventHandler(this._handle_blur_4_2.bind(this)));
    var disposable_5:Function = this.renderer.listen(this._el_4,'focus',this.eventHandler(this._handle_focus_4_3.bind(this)));
    this._expr_13 = import7.UNINITIALIZED;
    const subscription_0:any = this._NgModel_4_5.update.subscribe(this.eventHandler(this._handle_ngModelChange_4_0.bind(this)));
    this._expr_14 = import7.UNINITIALIZED;
    this._expr_15 = import7.UNINITIALIZED;
    this._expr_16 = import7.UNINITIALIZED;
    this._expr_17 = import7.UNINITIALIZED;
    this._expr_18 = import7.UNINITIALIZED;
    this._expr_19 = import7.UNINITIALIZED;
    var disposable_6:Function = this.renderer.listen(this._el_5,'click',this.eventHandler(this._handle_click_5_0.bind(this)));
    var disposable_7:Function = this.renderer.listen(this._el_5,'mousedown',this.eventHandler(this._handle_mousedown_5_1.bind(this)));
    this._expr_22 = import7.UNINITIALIZED;
    this._expr_25 = import7.UNINITIALIZED;
    var disposable_8:Function = this.renderer.listen(this._el_6,'click',this.eventHandler(this._handle_click_6_0.bind(this)));
    var disposable_9:Function = this.renderer.listen(this._el_6,'mousedown',this.eventHandler(this._handle_mousedown_6_1.bind(this)));
    this._expr_26 = import7.UNINITIALIZED;
    this._expr_27 = import7.UNINITIALIZED;
    this._viewQuery_searchbarInput_0.reset([new import9.ElementRef(this._el_4)]);
    this.context.searchbarInput = this._viewQuery_searchbarInput_0.first;
    this._viewQuery_searchbarIcon_1.reset([new import9.ElementRef(this._el_3)]);
    this.context._searchbarIcon = this._viewQuery_searchbarIcon_1.first;
    this._viewQuery_cancelButton_2.reset([new import9.ElementRef(this._el_6)]);
    this.context._cancelButton = this._viewQuery_cancelButton_2.first;
    this.init([],[
      this._el_0,
      this._el_1,
      this._el_2,
      this._el_3,
      this._el_4,
      this._el_5,
      this._el_6,
      this._text_7
    ]
    ,[
      disposable_0,
      disposable_1,
      disposable_2,
      disposable_3,
      disposable_4,
      disposable_5,
      disposable_6,
      disposable_7,
      disposable_8,
      disposable_9
    ]
    ,[subscription_0]);
    return (null as any);
  }
  injectorGetInternal(token:any,requestNodeIndex:number,notFoundResult:any):any {
    if (((token === import15.Icon) && (2 === requestNodeIndex))) { return this._Icon_2_3; }
    if (((token === import14.Button) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) { return this._Button_1_4; }
    if (((token === import16.DefaultValueAccessor) && (4 === requestNodeIndex))) { return this._DefaultValueAccessor_4_3; }
    if (((token === import20.NG_VALUE_ACCESSOR) && (4 === requestNodeIndex))) { return this._NG_VALUE_ACCESSOR_4_4; }
    if (((token === import17.NgModel) && (4 === requestNodeIndex))) { return this._NgModel_4_5; }
    if (((token === import10.NgControl) && (4 === requestNodeIndex))) { return this._NgControl_4_6; }
    if (((token === import18.NgControlStatus) && (4 === requestNodeIndex))) { return this._NgControlStatus_4_7; }
    if (((token === import14.Button) && (5 === requestNodeIndex))) { return this._Button_5_4; }
    if (((token === import14.Button) && ((6 <= requestNodeIndex) && (requestNodeIndex <= 7)))) { return this._Button_6_4; }
    return notFoundResult;
  }
  detectChangesInternal(throwOnChange:boolean):void {
    var changed:boolean = true;
    var changes:{[key: string]:import7.SimpleChange} = (null as any);
    changed = false;
    const currVal_2:any = '';
    if (import4.checkBinding(throwOnChange,this._expr_2,currVal_2)) {
      this._Button_1_4.clear = currVal_2;
      changed = true;
      this._expr_2 = currVal_2;
    }
    const currVal_3:any = 'dark';
    if (import4.checkBinding(throwOnChange,this._expr_3,currVal_3)) {
      this._Button_1_4.color = currVal_3;
      changed = true;
      this._expr_3 = currVal_3;
    }
    if (changed) { this._appEl_1.componentView.markAsCheckOnce(); }
    const currVal_4:any = 'arrow-back';
    if (import4.checkBinding(throwOnChange,this._expr_4,currVal_4)) {
      this._Icon_2_3.name = currVal_4;
      this._expr_4 = currVal_4;
    }
    changes = (null as any);
    const currVal_13:any = this.context._value;
    if (import4.checkBinding(throwOnChange,this._expr_13,currVal_13)) {
      this._NgModel_4_5.model = currVal_13;
      if ((changes === (null as any))) { (changes = {}); }
      changes['model'] = new import7.SimpleChange(this._expr_13,currVal_13);
      this._expr_13 = currVal_13;
    }
    if ((changes !== (null as any))) { this._NgModel_4_5.ngOnChanges(changes); }
    changed = false;
    const currVal_22:any = '';
    if (import4.checkBinding(throwOnChange,this._expr_22,currVal_22)) {
      this._Button_5_4.clear = currVal_22;
      changed = true;
      this._expr_22 = currVal_22;
    }
    if (changed) { this._appEl_5.componentView.markAsCheckOnce(); }
    changed = false;
    const currVal_26:any = '';
    if (import4.checkBinding(throwOnChange,this._expr_26,currVal_26)) {
      this._Button_6_4.clear = currVal_26;
      changed = true;
      this._expr_26 = currVal_26;
    }
    if (changed) { this._appEl_6.componentView.markAsCheckOnce(); }
    this.detectContentChildrenChanges(throwOnChange);
    if (!throwOnChange) {
      if ((this.numberOfChecks === 0)) { this._Button_1_4.ngAfterContentInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_5_4.ngAfterContentInit(); }
      if ((this.numberOfChecks === 0)) { this._Button_6_4.ngAfterContentInit(); }
    }
    const currVal_5:any = this._Icon_2_3._hidden;
    if (import4.checkBinding(throwOnChange,this._expr_5,currVal_5)) {
      this.renderer.setElementClass(this._el_2,'hide',currVal_5);
      this._expr_5 = currVal_5;
    }
    const currVal_12:any = this.context.placeholder;
    if (import4.checkBinding(throwOnChange,this._expr_12,currVal_12)) {
      this.renderer.setElementAttribute(this._el_4,'placeholder',((currVal_12 == (null as any))? (null as any): currVal_12.toString()));
      this._expr_12 = currVal_12;
    }
    const currVal_14:any = this._NgControlStatus_4_7.ngClassUntouched;
    if (import4.checkBinding(throwOnChange,this._expr_14,currVal_14)) {
      this.renderer.setElementClass(this._el_4,'ng-untouched',currVal_14);
      this._expr_14 = currVal_14;
    }
    const currVal_15:any = this._NgControlStatus_4_7.ngClassTouched;
    if (import4.checkBinding(throwOnChange,this._expr_15,currVal_15)) {
      this.renderer.setElementClass(this._el_4,'ng-touched',currVal_15);
      this._expr_15 = currVal_15;
    }
    const currVal_16:any = this._NgControlStatus_4_7.ngClassPristine;
    if (import4.checkBinding(throwOnChange,this._expr_16,currVal_16)) {
      this.renderer.setElementClass(this._el_4,'ng-pristine',currVal_16);
      this._expr_16 = currVal_16;
    }
    const currVal_17:any = this._NgControlStatus_4_7.ngClassDirty;
    if (import4.checkBinding(throwOnChange,this._expr_17,currVal_17)) {
      this.renderer.setElementClass(this._el_4,'ng-dirty',currVal_17);
      this._expr_17 = currVal_17;
    }
    const currVal_18:any = this._NgControlStatus_4_7.ngClassValid;
    if (import4.checkBinding(throwOnChange,this._expr_18,currVal_18)) {
      this.renderer.setElementClass(this._el_4,'ng-valid',currVal_18);
      this._expr_18 = currVal_18;
    }
    const currVal_19:any = this._NgControlStatus_4_7.ngClassInvalid;
    if (import4.checkBinding(throwOnChange,this._expr_19,currVal_19)) {
      this.renderer.setElementClass(this._el_4,'ng-invalid',currVal_19);
      this._expr_19 = currVal_19;
    }
    const currVal_25:any = (this.context._isActive? 1: (0 - 1));
    if (import4.checkBinding(throwOnChange,this._expr_25,currVal_25)) {
      this.renderer.setElementProperty(this._el_6,'tabIndex',currVal_25);
      this._expr_25 = currVal_25;
    }
    const currVal_27:any = import4.interpolate(1,'',this.context.cancelButtonText,'');
    if (import4.checkBinding(throwOnChange,this._expr_27,currVal_27)) {
      this.renderer.setText(this._text_7,currVal_27);
      this._expr_27 = currVal_27;
    }
    this.detectViewChildrenChanges(throwOnChange);
  }
  destroyInternal():void {
    this._Icon_2_3.ngOnDestroy();
    this._NgModel_4_5.ngOnDestroy();
  }
  private _handle_click_1_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.cancelSearchbar($event)) !== false);
    return (true && pd_0);
  }
  private _handle_mousedown_1_1($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.cancelSearchbar($event)) !== false);
    return (true && pd_0);
  }
  private _handle_ngModelChange_4_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>(this.context._value = $event)) !== false);
    return (true && pd_0);
  }
  private _handle_input_4_1($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.inputChanged($event)) !== false);
    const pd_1:any = ((<any>this._DefaultValueAccessor_4_3.onChange($event.target.value)) !== false);
    return ((true && pd_0) && pd_1);
  }
  private _handle_blur_4_2($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.inputBlurred($event)) !== false);
    const pd_1:any = ((<any>this._DefaultValueAccessor_4_3.onTouched()) !== false);
    return ((true && pd_0) && pd_1);
  }
  private _handle_focus_4_3($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.inputFocused($event)) !== false);
    return (true && pd_0);
  }
  private _handle_click_5_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.clearInput($event)) !== false);
    return (true && pd_0);
  }
  private _handle_mousedown_5_1($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.clearInput($event)) !== false);
    return (true && pd_0);
  }
  private _handle_click_6_0($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.cancelSearchbar($event)) !== false);
    return (true && pd_0);
  }
  private _handle_mousedown_6_1($event:any):boolean {
    this.markPathToRootAsCheckOnce();
    const pd_0:any = ((<any>this.context.cancelSearchbar($event)) !== false);
    return (true && pd_0);
  }
}
export function viewFactory_Searchbar0(viewUtils:import4.ViewUtils,parentInjector:import5.Injector,declarationEl:import2.AppElement):import1.AppView<import3.Searchbar> {
  if ((renderType_Searchbar === (null as any))) { (renderType_Searchbar = viewUtils.createRenderComponentType('',0,import11.ViewEncapsulation.None,styles_Searchbar,{})); }
  return new _View_Searchbar0(viewUtils,parentInjector,declarationEl);
}
