// ionic-angular/components/searchbar/searchbar.ngfactory.js
/**
 * This file is generated by the Angular 2 template compiler.
 * Do not edit.
 */
/* tslint:disable */
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
import * as import1 from '@angular/core/src/linker/view';
import * as import2 from '@angular/core/src/linker/element';
import * as import3 from 'ionic-angular/components/searchbar/searchbar';
import * as import4 from '@angular/core/src/linker/view_utils';
import * as import6 from '@angular/core/src/linker/view_type';
import * as import7 from '@angular/core/src/change_detection/change_detection';
import * as import8 from 'ionic-angular/config/config';
import * as import9 from '@angular/core/src/linker/element_ref';
import * as import10 from '@angular/forms/src/directives/ng_control';
import * as import11 from '@angular/core/src/metadata/view';
import * as import12 from '@angular/core/src/linker/component_factory';
import * as import13 from '@angular/core/src/linker/query_list';
import * as import14 from 'ionic-angular/components/button/button';
import * as import15 from 'ionic-angular/components/icon/icon';
import * as import16 from '@angular/forms/src/directives/default_value_accessor';
import * as import17 from '@angular/forms/src/directives/ng_model';
import * as import18 from '@angular/forms/src/directives/ng_control_status';
import * as import19 from '../button/button.ngfactory';
import * as import20 from '@angular/forms/src/directives/control_value_accessor';
var renderType_Searchbar_Host = null;
var _View_Searchbar_Host0 = (function (_super) {
    __extends(_View_Searchbar_Host0, _super);
    function _View_Searchbar_Host0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Searchbar_Host0, renderType_Searchbar_Host, import6.ViewType.HOST, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Searchbar_Host0.prototype.createInternal = function (rootSelector) {
        this._el_0 = this.selectOrCreateHostElement('ion-searchbar', rootSelector, null);
        this._appEl_0 = new import2.AppElement(0, null, this, this._el_0);
        var compView_0 = viewFactory_Searchbar0(this.viewUtils, this.injector(0), this._appEl_0);
        this._Searchbar_0_4 = new import3.Searchbar(this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_0), this.renderer, this.parentInjector.get(import10.NgControl, null));
        this._appEl_0.initComponent(this._Searchbar_0_4, [], compView_0);
        compView_0.create(this._Searchbar_0_4, this.projectableNodes, null);
        this._expr_0 = import7.UNINITIALIZED;
        this._expr_1 = import7.UNINITIALIZED;
        this._expr_2 = import7.UNINITIALIZED;
        this._expr_3 = import7.UNINITIALIZED;
        this._expr_4 = import7.UNINITIALIZED;
        this.init([].concat([this._el_0]), [this._el_0], [], []);
        return this._appEl_0;
    };
    _View_Searchbar_Host0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import3.Searchbar) && (0 === requestNodeIndex))) {
            return this._Searchbar_0_4;
        }
        return notFoundResult;
    };
    _View_Searchbar_Host0.prototype.detectChangesInternal = function (throwOnChange) {
        if (((this.numberOfChecks === 0) && !throwOnChange)) {
            this._Searchbar_0_4.ngOnInit();
        }
        this.detectContentChildrenChanges(throwOnChange);
        var currVal_0 = this._Searchbar_0_4._value;
        if (import4.checkBinding(throwOnChange, this._expr_0, currVal_0)) {
            this.renderer.setElementClass(this._el_0, 'searchbar-has-value', currVal_0);
            this._expr_0 = currVal_0;
        }
        var currVal_1 = this._Searchbar_0_4._isActive;
        if (import4.checkBinding(throwOnChange, this._expr_1, currVal_1)) {
            this.renderer.setElementClass(this._el_0, 'searchbar-active', currVal_1);
            this._expr_1 = currVal_1;
        }
        var currVal_2 = this._Searchbar_0_4.showCancelButton;
        if (import4.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this.renderer.setElementClass(this._el_0, 'searchbar-show-cancel', currVal_2);
            this._expr_2 = currVal_2;
        }
        var currVal_3 = this._Searchbar_0_4.shouldAlignLeft();
        if (import4.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this.renderer.setElementClass(this._el_0, 'searchbar-left-aligned', currVal_3);
            this._expr_3 = currVal_3;
        }
        var currVal_4 = this._Searchbar_0_4._sbHasFocus;
        if (import4.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this.renderer.setElementClass(this._el_0, 'searchbar-has-focus', currVal_4);
            this._expr_4 = currVal_4;
        }
        this.detectViewChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            this._Searchbar_0_4.ngAfterViewChecked();
        }
    };
    return _View_Searchbar_Host0;
}(import1.AppView));
function viewFactory_Searchbar_Host0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Searchbar_Host === null)) {
        (renderType_Searchbar_Host = viewUtils.createRenderComponentType('', 0, import11.ViewEncapsulation.None, [], {}));
    }
    return new _View_Searchbar_Host0(viewUtils, parentInjector, declarationEl);
}
export var SearchbarNgFactory = new import12.ComponentFactory('ion-searchbar', viewFactory_Searchbar_Host0, import3.Searchbar);
var styles_Searchbar = [];
var renderType_Searchbar = null;
var _View_Searchbar0 = (function (_super) {
    __extends(_View_Searchbar0, _super);
    function _View_Searchbar0(viewUtils, parentInjector, declarationEl) {
        _super.call(this, _View_Searchbar0, renderType_Searchbar, import6.ViewType.COMPONENT, viewUtils, parentInjector, declarationEl, import7.ChangeDetectorStatus.CheckAlways);
    }
    _View_Searchbar0.prototype.createInternal = function (rootSelector) {
        var parentRenderNode = this.renderer.createViewRoot(this.declarationAppElement.nativeElement);
        this._viewQuery_searchbarInput_0 = new import13.QueryList();
        this._viewQuery_searchbarIcon_1 = new import13.QueryList();
        this._viewQuery_cancelButton_2 = new import13.QueryList();
        this._el_0 = this.renderer.createElement(parentRenderNode, 'div', null);
        this.renderer.setElementAttribute(this._el_0, 'class', 'searchbar-input-container');
        this._el_1 = this.renderer.createElement(this._el_0, 'button', null);
        this.renderer.setElementAttribute(this._el_1, 'class', 'searchbar-md-cancel');
        this.renderer.setElementAttribute(this._el_1, 'clear', '');
        this.renderer.setElementAttribute(this._el_1, 'color', 'dark');
        this.renderer.setElementAttribute(this._el_1, 'ion-button', '');
        this._appEl_1 = new import2.AppElement(1, 0, this, this._el_1);
        var compView_1 = import19.viewFactory_Button0(this.viewUtils, this.injector(1), this._appEl_1);
        this._Button_1_4 = new import14.Button(null, '', this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_1), this.renderer);
        this._appEl_1.initComponent(this._Button_1_4, [], compView_1);
        this._el_2 = this.renderer.createElement(null, 'ion-icon', null);
        this.renderer.setElementAttribute(this._el_2, 'name', 'arrow-back');
        this.renderer.setElementAttribute(this._el_2, 'role', 'img');
        this._Icon_2_3 = new import15.Icon(this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_2), this.renderer);
        compView_1.create(this._Button_1_4, [[].concat([this._el_2])], null);
        this._el_3 = this.renderer.createElement(this._el_0, 'div', null);
        this.renderer.setElementAttribute(this._el_3, 'class', 'searchbar-search-icon');
        this._el_4 = this.renderer.createElement(this._el_0, 'input', null);
        this.renderer.setElementAttribute(this._el_4, 'class', 'searchbar-input');
        this._DefaultValueAccessor_4_3 = new import16.DefaultValueAccessor(this.renderer, new import9.ElementRef(this._el_4));
        this._NG_VALUE_ACCESSOR_4_4 = [this._DefaultValueAccessor_4_3];
        this._NgModel_4_5 = new import17.NgModel(null, null, null, this._NG_VALUE_ACCESSOR_4_4);
        this._NgControl_4_6 = this._NgModel_4_5;
        this._NgControlStatus_4_7 = new import18.NgControlStatus(this._NgControl_4_6);
        this._el_5 = this.renderer.createElement(this._el_0, 'button', null);
        this.renderer.setElementAttribute(this._el_5, 'class', 'searchbar-clear-icon');
        this.renderer.setElementAttribute(this._el_5, 'clear', '');
        this.renderer.setElementAttribute(this._el_5, 'ion-button', '');
        this._appEl_5 = new import2.AppElement(5, 0, this, this._el_5);
        var compView_5 = import19.viewFactory_Button0(this.viewUtils, this.injector(5), this._appEl_5);
        this._Button_5_4 = new import14.Button(null, '', this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_5), this.renderer);
        this._appEl_5.initComponent(this._Button_5_4, [], compView_5);
        compView_5.create(this._Button_5_4, [[]], null);
        this._el_6 = this.renderer.createElement(parentRenderNode, 'button', null);
        this.renderer.setElementAttribute(this._el_6, 'class', 'searchbar-ios-cancel');
        this.renderer.setElementAttribute(this._el_6, 'clear', '');
        this.renderer.setElementAttribute(this._el_6, 'ion-button', '');
        this._appEl_6 = new import2.AppElement(6, null, this, this._el_6);
        var compView_6 = import19.viewFactory_Button0(this.viewUtils, this.injector(6), this._appEl_6);
        this._Button_6_4 = new import14.Button(null, '', this.parentInjector.get(import8.Config), new import9.ElementRef(this._el_6), this.renderer);
        this._appEl_6.initComponent(this._Button_6_4, [], compView_6);
        this._text_7 = this.renderer.createText(null, '', null);
        compView_6.create(this._Button_6_4, [[].concat([this._text_7])], null);
        var disposable_0 = this.renderer.listen(this._el_1, 'click', this.eventHandler(this._handle_click_1_0.bind(this)));
        var disposable_1 = this.renderer.listen(this._el_1, 'mousedown', this.eventHandler(this._handle_mousedown_1_1.bind(this)));
        this._expr_2 = import7.UNINITIALIZED;
        this._expr_3 = import7.UNINITIALIZED;
        this._expr_4 = import7.UNINITIALIZED;
        this._expr_5 = import7.UNINITIALIZED;
        this._expr_12 = import7.UNINITIALIZED;
        var disposable_2 = this.renderer.listen(this._el_4, 'ngModelChange', this.eventHandler(this._handle_ngModelChange_4_0.bind(this)));
        var disposable_3 = this.renderer.listen(this._el_4, 'input', this.eventHandler(this._handle_input_4_1.bind(this)));
        var disposable_4 = this.renderer.listen(this._el_4, 'blur', this.eventHandler(this._handle_blur_4_2.bind(this)));
        var disposable_5 = this.renderer.listen(this._el_4, 'focus', this.eventHandler(this._handle_focus_4_3.bind(this)));
        this._expr_13 = import7.UNINITIALIZED;
        var subscription_0 = this._NgModel_4_5.update.subscribe(this.eventHandler(this._handle_ngModelChange_4_0.bind(this)));
        this._expr_14 = import7.UNINITIALIZED;
        this._expr_15 = import7.UNINITIALIZED;
        this._expr_16 = import7.UNINITIALIZED;
        this._expr_17 = import7.UNINITIALIZED;
        this._expr_18 = import7.UNINITIALIZED;
        this._expr_19 = import7.UNINITIALIZED;
        var disposable_6 = this.renderer.listen(this._el_5, 'click', this.eventHandler(this._handle_click_5_0.bind(this)));
        var disposable_7 = this.renderer.listen(this._el_5, 'mousedown', this.eventHandler(this._handle_mousedown_5_1.bind(this)));
        this._expr_22 = import7.UNINITIALIZED;
        this._expr_25 = import7.UNINITIALIZED;
        var disposable_8 = this.renderer.listen(this._el_6, 'click', this.eventHandler(this._handle_click_6_0.bind(this)));
        var disposable_9 = this.renderer.listen(this._el_6, 'mousedown', this.eventHandler(this._handle_mousedown_6_1.bind(this)));
        this._expr_26 = import7.UNINITIALIZED;
        this._expr_27 = import7.UNINITIALIZED;
        this._viewQuery_searchbarInput_0.reset([new import9.ElementRef(this._el_4)]);
        this.context.searchbarInput = this._viewQuery_searchbarInput_0.first;
        this._viewQuery_searchbarIcon_1.reset([new import9.ElementRef(this._el_3)]);
        this.context._searchbarIcon = this._viewQuery_searchbarIcon_1.first;
        this._viewQuery_cancelButton_2.reset([new import9.ElementRef(this._el_6)]);
        this.context._cancelButton = this._viewQuery_cancelButton_2.first;
        this.init([], [
            this._el_0,
            this._el_1,
            this._el_2,
            this._el_3,
            this._el_4,
            this._el_5,
            this._el_6,
            this._text_7
        ], [
            disposable_0,
            disposable_1,
            disposable_2,
            disposable_3,
            disposable_4,
            disposable_5,
            disposable_6,
            disposable_7,
            disposable_8,
            disposable_9
        ], [subscription_0]);
        return null;
    };
    _View_Searchbar0.prototype.injectorGetInternal = function (token, requestNodeIndex, notFoundResult) {
        if (((token === import15.Icon) && (2 === requestNodeIndex))) {
            return this._Icon_2_3;
        }
        if (((token === import14.Button) && ((1 <= requestNodeIndex) && (requestNodeIndex <= 2)))) {
            return this._Button_1_4;
        }
        if (((token === import16.DefaultValueAccessor) && (4 === requestNodeIndex))) {
            return this._DefaultValueAccessor_4_3;
        }
        if (((token === import20.NG_VALUE_ACCESSOR) && (4 === requestNodeIndex))) {
            return this._NG_VALUE_ACCESSOR_4_4;
        }
        if (((token === import17.NgModel) && (4 === requestNodeIndex))) {
            return this._NgModel_4_5;
        }
        if (((token === import10.NgControl) && (4 === requestNodeIndex))) {
            return this._NgControl_4_6;
        }
        if (((token === import18.NgControlStatus) && (4 === requestNodeIndex))) {
            return this._NgControlStatus_4_7;
        }
        if (((token === import14.Button) && (5 === requestNodeIndex))) {
            return this._Button_5_4;
        }
        if (((token === import14.Button) && ((6 <= requestNodeIndex) && (requestNodeIndex <= 7)))) {
            return this._Button_6_4;
        }
        return notFoundResult;
    };
    _View_Searchbar0.prototype.detectChangesInternal = function (throwOnChange) {
        var changed = true;
        var changes = null;
        changed = false;
        var currVal_2 = '';
        if (import4.checkBinding(throwOnChange, this._expr_2, currVal_2)) {
            this._Button_1_4.clear = currVal_2;
            changed = true;
            this._expr_2 = currVal_2;
        }
        var currVal_3 = 'dark';
        if (import4.checkBinding(throwOnChange, this._expr_3, currVal_3)) {
            this._Button_1_4.color = currVal_3;
            changed = true;
            this._expr_3 = currVal_3;
        }
        if (changed) {
            this._appEl_1.componentView.markAsCheckOnce();
        }
        var currVal_4 = 'arrow-back';
        if (import4.checkBinding(throwOnChange, this._expr_4, currVal_4)) {
            this._Icon_2_3.name = currVal_4;
            this._expr_4 = currVal_4;
        }
        changes = null;
        var currVal_13 = this.context._value;
        if (import4.checkBinding(throwOnChange, this._expr_13, currVal_13)) {
            this._NgModel_4_5.model = currVal_13;
            if ((changes === null)) {
                (changes = {});
            }
            changes['model'] = new import7.SimpleChange(this._expr_13, currVal_13);
            this._expr_13 = currVal_13;
        }
        if ((changes !== null)) {
            this._NgModel_4_5.ngOnChanges(changes);
        }
        changed = false;
        var currVal_22 = '';
        if (import4.checkBinding(throwOnChange, this._expr_22, currVal_22)) {
            this._Button_5_4.clear = currVal_22;
            changed = true;
            this._expr_22 = currVal_22;
        }
        if (changed) {
            this._appEl_5.componentView.markAsCheckOnce();
        }
        changed = false;
        var currVal_26 = '';
        if (import4.checkBinding(throwOnChange, this._expr_26, currVal_26)) {
            this._Button_6_4.clear = currVal_26;
            changed = true;
            this._expr_26 = currVal_26;
        }
        if (changed) {
            this._appEl_6.componentView.markAsCheckOnce();
        }
        this.detectContentChildrenChanges(throwOnChange);
        if (!throwOnChange) {
            if ((this.numberOfChecks === 0)) {
                this._Button_1_4.ngAfterContentInit();
            }
            if ((this.numberOfChecks === 0)) {
                this._Button_5_4.ngAfterContentInit();
            }
            if ((this.numberOfChecks === 0)) {
                this._Button_6_4.ngAfterContentInit();
            }
        }
        var currVal_5 = this._Icon_2_3._hidden;
        if (import4.checkBinding(throwOnChange, this._expr_5, currVal_5)) {
            this.renderer.setElementClass(this._el_2, 'hide', currVal_5);
            this._expr_5 = currVal_5;
        }
        var currVal_12 = this.context.placeholder;
        if (import4.checkBinding(throwOnChange, this._expr_12, currVal_12)) {
            this.renderer.setElementAttribute(this._el_4, 'placeholder', ((currVal_12 == null) ? null : currVal_12.toString()));
            this._expr_12 = currVal_12;
        }
        var currVal_14 = this._NgControlStatus_4_7.ngClassUntouched;
        if (import4.checkBinding(throwOnChange, this._expr_14, currVal_14)) {
            this.renderer.setElementClass(this._el_4, 'ng-untouched', currVal_14);
            this._expr_14 = currVal_14;
        }
        var currVal_15 = this._NgControlStatus_4_7.ngClassTouched;
        if (import4.checkBinding(throwOnChange, this._expr_15, currVal_15)) {
            this.renderer.setElementClass(this._el_4, 'ng-touched', currVal_15);
            this._expr_15 = currVal_15;
        }
        var currVal_16 = this._NgControlStatus_4_7.ngClassPristine;
        if (import4.checkBinding(throwOnChange, this._expr_16, currVal_16)) {
            this.renderer.setElementClass(this._el_4, 'ng-pristine', currVal_16);
            this._expr_16 = currVal_16;
        }
        var currVal_17 = this._NgControlStatus_4_7.ngClassDirty;
        if (import4.checkBinding(throwOnChange, this._expr_17, currVal_17)) {
            this.renderer.setElementClass(this._el_4, 'ng-dirty', currVal_17);
            this._expr_17 = currVal_17;
        }
        var currVal_18 = this._NgControlStatus_4_7.ngClassValid;
        if (import4.checkBinding(throwOnChange, this._expr_18, currVal_18)) {
            this.renderer.setElementClass(this._el_4, 'ng-valid', currVal_18);
            this._expr_18 = currVal_18;
        }
        var currVal_19 = this._NgControlStatus_4_7.ngClassInvalid;
        if (import4.checkBinding(throwOnChange, this._expr_19, currVal_19)) {
            this.renderer.setElementClass(this._el_4, 'ng-invalid', currVal_19);
            this._expr_19 = currVal_19;
        }
        var currVal_25 = (this.context._isActive ? 1 : (0 - 1));
        if (import4.checkBinding(throwOnChange, this._expr_25, currVal_25)) {
            this.renderer.setElementProperty(this._el_6, 'tabIndex', currVal_25);
            this._expr_25 = currVal_25;
        }
        var currVal_27 = import4.interpolate(1, '', this.context.cancelButtonText, '');
        if (import4.checkBinding(throwOnChange, this._expr_27, currVal_27)) {
            this.renderer.setText(this._text_7, currVal_27);
            this._expr_27 = currVal_27;
        }
        this.detectViewChildrenChanges(throwOnChange);
    };
    _View_Searchbar0.prototype.destroyInternal = function () {
        this._Icon_2_3.ngOnDestroy();
        this._NgModel_4_5.ngOnDestroy();
    };
    _View_Searchbar0.prototype._handle_click_1_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.cancelSearchbar($event) !== false);
        return (true && pd_0);
    };
    _View_Searchbar0.prototype._handle_mousedown_1_1 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.cancelSearchbar($event) !== false);
        return (true && pd_0);
    };
    _View_Searchbar0.prototype._handle_ngModelChange_4_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = ((this.context._value = $event) !== false);
        return (true && pd_0);
    };
    _View_Searchbar0.prototype._handle_input_4_1 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.inputChanged($event) !== false);
        var pd_1 = (this._DefaultValueAccessor_4_3.onChange($event.target.value) !== false);
        return ((true && pd_0) && pd_1);
    };
    _View_Searchbar0.prototype._handle_blur_4_2 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.inputBlurred($event) !== false);
        var pd_1 = (this._DefaultValueAccessor_4_3.onTouched() !== false);
        return ((true && pd_0) && pd_1);
    };
    _View_Searchbar0.prototype._handle_focus_4_3 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.inputFocused($event) !== false);
        return (true && pd_0);
    };
    _View_Searchbar0.prototype._handle_click_5_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.clearInput($event) !== false);
        return (true && pd_0);
    };
    _View_Searchbar0.prototype._handle_mousedown_5_1 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.clearInput($event) !== false);
        return (true && pd_0);
    };
    _View_Searchbar0.prototype._handle_click_6_0 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.cancelSearchbar($event) !== false);
        return (true && pd_0);
    };
    _View_Searchbar0.prototype._handle_mousedown_6_1 = function ($event) {
        this.markPathToRootAsCheckOnce();
        var pd_0 = (this.context.cancelSearchbar($event) !== false);
        return (true && pd_0);
    };
    return _View_Searchbar0;
}(import1.AppView));
export function viewFactory_Searchbar0(viewUtils, parentInjector, declarationEl) {
    if ((renderType_Searchbar === null)) {
        (renderType_Searchbar = viewUtils.createRenderComponentType('', 0, import11.ViewEncapsulation.None, styles_Searchbar, {}));
    }
    return new _View_Searchbar0(viewUtils, parentInjector, declarationEl);
}
